hbase-client/src/main/java/org/apache/hadoop/hbase/filter/PrefixFilter.java

```diff
@@ -77,7 +77,7 @@ public boolean filterRowKey(Cell firstRowCell) {
     if ((!isReversed() && cmp > 0) || (isReversed() && cmp < 0)) {
       passedPrefix = true;
     }
-    filterRow = (!isReversed() && cmp > 0) || (isReversed() && cmp < 0);
+    filterRow = (!isReversed() && cmp > 0) || (isReversed() && cmp != 0);
```

Comment:
This change looks fishy

Reply:
Completely agree. But this fixes the previously failing TestFilter.testPrefixFilterWithReverseScan unit test. :sweat_smile: